File: backend/src/main/java/com/apocalypse/thefall/repository/character/stats/SkillRepository.java
package com.apocalypse.thefall.repository.character.stats;

import com.apocalypse.thefall.entity.character.stats.Skill;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface SkillRepository extends JpaRepository<Skill, Long> {

    List<Skill> findAllByVisibleTrueOrderByDisplayOrderAsc();
}

File: backend/src/main/java/com/apocalypse/thefall/repository/character/stats/SkillRepository.kt
package com.apocalypse.thefall.repository.character.stats

import com.apocalypse.thefall.entity.character.stats.Skill
import jakarta.persistence.QueryHint
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.QueryHints

interface SkillRepository : JpaRepository<Skill, Long> {
    @QueryHints(QueryHint(name = "org.hibernate.cacheable", value = "true"))
    fun findAllByVisibleTrueOrderByDisplayOrderAsc(): List<Skill>
}
